WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/WooShipping Create Shipping Labels/WooShipping Split Shipments/CollapsibleShipmentCardViewModel.swift

```diff
@@ -0,0 +1,60 @@
+import SwiftUI
+import WooFoundation
+import Yosemite
+
+/// View model for `CollapsibleShipmentCard`.
+final class CollapsibleShipmentCardViewModel: ObservableObject, Identifiable {
+    let id = UUID()
+
+    /// The main shipment row.
+    let mainShipmentRow: SelectableShipmentRowViewModel
+
+    /// Child shipment rows, if the shipment has more than one quantity
+    let childShipmentRows: [SelectableShipmentRowViewModel]
+
+    init(parentShipmentId: String,
+         childShipmentIds: [String],
+         item: ShippingLabelPackageItem,
+         currency: String) {
+        let mainShippingItem = WooShippingItemRowViewModel(item: ShippingLabelPackageItem(copy: item, quantity: max(1.0, Decimal(childShipmentIds.count))),
+                                                           currency: currency)
+        let childShippingItem = WooShippingItemRowViewModel(item: ShippingLabelPackageItem(copy: item, quantity: 1.0),
+                                                            currency: currency)
+
+        self.mainShipmentRow = SelectableShipmentRowViewModel(shipmentId: parentShipmentId,
+                                                              isSelectable: true,
+                                                              item: mainShippingItem,
+                                                              showQuantity: true)
+        self.childShipmentRows = childShipmentIds.map({
+            SelectableShipmentRowViewModel(shipmentId: $0,
+                                           isSelectable: true,
+                                           item: childShippingItem,
+                                           showQuantity: false)
+        })
+
+        observeSelection()
+    }
+
+    func selectAll() {
+        mainShipmentRow.setSelected(true)
+        childShipmentRows.forEach({ $0.setSelected(true) })
+    }
+}
+
+private extension CollapsibleShipmentCardViewModel {
+    func observeSelection() {
+        self.mainShipmentRow.onSelectedChange = { [weak self] row in
+            guard let self else { return }
+
+            self.childShipmentRows.forEach({ $0.setSelected(row.selected) })
+        }
+
+        self.childShipmentRows.forEach({
+            $0.onSelectedChange = { [weak self] row in
+                guard let self else { return }
+
+                self.mainShipmentRow.setSelected(false)
+            }
+        })
```

Comment:
Super nit:

```suggestion
        childShipmentRows.forEach({
            $0.onSelectedChange = { [weak self] row in
                guard let self else { return }

                mainShipmentRow.setSelected(false)
            }
        })
```

Reply:
Done in https://github.com/woocommerce/woocommerce-ios/pull/15397/commits/0cdc5a3e86fd228a6576dc5559444115a206d1b6